syntax = "proto3";

package mediocre.image.ocr.v1beta;

import "mediocre/image/v1beta/image.proto";

service OcrService {
  rpc GetCharacters(GetCharactersRequest) returns (GetCharactersResponse);
}

message GetCharactersRequest {
  image.v1beta.Image image = 1;
  GetCharactersParams params = 2;
}

message GetCharactersParams {
  oneof params {
    TesseractParams tesseract_params = 1;
    MediocreParams mediocre_params = 2;
  }
}

message TesseractParams {}

message MediocreParams {
  oneof method {
    NumericDisplayParams numeric_display_params = 1;
  }
}

message NumericDisplayParams {}

message GetCharactersResponse {
  string characters = 1;
}
